### DISCUSSION TITLE
rules_docker needs maintainers

### DISCUSSION THREAD

**Discussion by alexeagle** [bazelbuild/rules_docker#2038] [Announcements]
The current three maintainers, @pcj @gravypod and @alexeagle don't have enough spare time recently to maintain rules_docker.

Currently the rules are in a minimal maintenance mode where only high-severity issues get attention, when one of us is pinged on Slack.

We hope that some corporate user who makes heavy use of rules_docker will step up and fund the maintenance work with part-time engineering or contributions to the rules authors SIG.

**Comment by numine777** [comment]
I can say that I'm a relative novice with regards to bazel, but if I can help in any way (ideally beginning with more trivial issues and working towards more advanced), I'm happy to lend what help I can.

**Comment by alexeagle** [comment]
@uhthomas you've been active in the repo recently. Do you have time/interest in taking a role like this, or does your organization have resources to help out?

**Reply by uhthomas** [reply]
I'm not sure my organisation currently has the resource to help out at current unfortunately.

I'd be more happy to become a maintainer of rules_docker as an individual however :)

**Reply by gravypod** [reply]
@uhthomas if you could either reach out to me on the Bazel slack or via email I can invite you to our weekly sync meeting for the project. We can talk about next steps!

**Comment by linzhp** [comment]
We have a moderate usage of rules_docker at Uber (95 `container_image` targets, 25 `container_pull`, 5 `container_push` targets) in Go monorepo. I can help a bit by reviewing some Bazel/Go related pull requests and testing major changes in Uber, but my knowledge about docker is limited.

**Reply by gravypod** [reply]
@linzhp  if you could either reach out to me on the Bazel slack or via email I can invite you to our weekly sync meeting for the project. We can talk about next steps!

**Comment by alexeagle** [comment]
@gravypod do you want to get the admins for this repo to add @uhthomas and @linzhp as maintainers, to at least be able to review/approve PRs here?

**Comment by alexeagle** [comment]
Thanks for volunteering @linzhp and @uhthomas !

**Reply by nvachhar** [reply]
I'm a bit late to the show. But at work, I'd like to start using this more, so I have a vested interest in it not being on life support. I can lend a hand as well.

**Comment by alexeagle** [comment]
Nearly a year on, I am very thankful for the help from @linzhp and @uhthomas to "keep the lights on".
However I think users are probably getting wrong expectations about the health of this project.
I think we should add something to the README explaining that rules_docker is on minimal life support and users should not expect issues to be resolved, no new features, unlikely to review large PRs, etc. What do the current maintainers think?

**Reply by uhthomas** [reply]
@alexeagle That might be a reasonable thing to do. Aside from being busy, I've also felt a little directionless with respect to rules_docker given many changes in the general OCI space and alternatives like rules_oci. Really we need to have a formal discussion about the future of this project and create a roadmap, if any.

**Reply by alexeagle** [reply]
Is this the place for the formal discussion? Or would you rather have a meeting, and who should be invited? @meteorcloudy and @radvani13 already in the loop about the need for this.

And, do you want to finish that formal discussion before making an edit to the README, or should we put a preliminary one now? I'm thinking something like

"rules_docker is in minimal maintenance mode. Issues are unlikely to be addressed. We'll review critical PRs such as security fixes, but other PRs may not be reviewed. The maintainers are determining if the project has a long-term future, and hope to post a roadmap soon."

**Reply by seh** [reply]
Please count me in on such a meeting.

**Reply by linzhp** [reply]
I am fine with putting the preliminary one out now and update it after the meeting if necessary.

**Reply by alexeagle** [reply]
The only activity in the last couple months on rules_docker is from stalebot asking if 6mo-neglected issues should be closed.
https://bazel.build/rules#recommended-rules now lists rules_oci instead.

@uhthomas @linzhp @pcj @gravypod as the remaining maintainers here, what would you like to do? I think options include:

1. Publish the "roadmap" that "we hope to post" according to https://github.com/bazelbuild/rules_docker#status
2. Turn off the stalebot and let the repo sit as-is, hoping that users figure out the situation on their own.
3. Archive this repo for now to send a clear signal to users. If someone decides to invest here, it's trivial to un-archive.

What do the four of you think?

**Comment by nvachhar** [comment]
Is `rules_oci` the official replacement?

**Reply by alexeagle** [reply]
I don't know what "official" means, Bazel doesn't have DevRel so no one from Google is likely to answer that.
rules_oci isn't a drop-in replacement as it has a smaller scope.

**Reply by alexeagle** [reply]
Still not "official" but it was posted by Google: https://security.googleblog.com/2023/05/introducing-rulesoci.html

**Reply by uhthomas** [reply]
@alexeagle Nice! Congratulations on the official(?) launch. Is there anything I can do to help bring rules_oci into a full replacement for rules_docker?

I saw you opened https://github.com/bazelbuild/rules_k8s/issues/723, it would be nice to address this, though rules_k8s seems to also be lacking a bit of love.

**Reply by alexeagle** [reply]
Thanks!
Yes, rules_k8s seems like a great place to help. I did some poking on bazel slack https://bazelbuild.slack.com/archives/CDCMRLS23/p1682523689503889 and the likely answer is that it has no maintainers. @chrislovecnm FYI

As for feature parity - we'll see what issues people raise for missing features. The main one I know of is https://github.com/bazel-contrib/rules_oci/issues/132.

**Reply by alexeagle** [reply]
According to https://bazel.build/rules#recommended-rules rules_oci is now the "official" recommendation.

**Comment by pcj** [comment]
Personally I feel it's premature to recommend rules_oci as an official replacement for rules_docker. 

Though this repo has problems attracting new maintainers, this is true for the majority of bazel rules repositories, rules_docker is not really an exception.

rules_docker has a fairly significant install base whereas it is unclear how many production users there are of rules_oci.  If we start seeing blog posts or other external signals were people are wholesale replacing rules_docker with rules_oci, maybe it would be different?

**Comment by alexeagle** [comment]
The repo was unarchived, I'm going to close this
